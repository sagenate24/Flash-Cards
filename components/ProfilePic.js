import React, { PureComponent } from 'react';
import { Image, Platform } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { connect } from 'react-redux';

class ProfilePic extends PureComponent {
  render() {
    const { styles, borderColor, backUpSize, profile, backUp } = this.props;

    if (backUp === false) {
      return <Image style={[styles, { borderColor }]} source={{ uri: profile.avatar }} />;
    }
    if (profile.avatar === '') {
      return (
        Platform.OS === 'ios'
          ? <Ionicons name="ios-person" size={backUpSize + 15} color={borderColor} />
          : <Ionicons name="md-person" size={backUpSize} color={borderColor} />
      );
    }

    return <Image style={[styles, { borderColor }]} source={{ uri: profile.avatar }} />;
  }
}

function mapStateToProps({ profile }) {
  return {
    profile,
  };
}

export default connect(mapStateToProps)(ProfilePic);
