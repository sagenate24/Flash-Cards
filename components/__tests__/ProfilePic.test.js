import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProfilePic from '../ProfilePic';

const mockProfile = {
  username: '',
  avatar: '',
  cover: '',
};

it('should match Snapshot', (done) => {
  const snap = renderer.create(
    <ProfilePic.WrappedComponent backUpSize={30} styles={{ width: 40, height: 40 }} borderColor='#fff' profile={mockProfile} />,
  ).toJSON();

  expect(snap).toMatchSnapshot();

  done();
});
