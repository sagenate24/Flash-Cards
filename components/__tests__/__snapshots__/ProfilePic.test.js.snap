// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
